ci: Use locally built images during testing

pmengelbert opened the pull request:
See issue #444 

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #444

pmengelbert committed:
Modify code to use specified pull policies

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Update tests to set args using new code

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
use test namespace

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Update tests to use proper namespace

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Simplify string code

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Satisfy linter

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

sozercan commented on main.go:
```diff
@@ -50,8 +52,11 @@ var (
-	profilePort   = flag.Int("pprof-port", 6060, "port for pprof profiling. defaulted to 6060 if unspecified")
+	enableProfile       = flag.Bool("enable-pprof", false, "enable pprof profiling")
+	profilePort         = flag.Int("pprof-port", 6060, "port for pprof profiling. defaulted to 6060 if unspecified")
+	eraserPullPolicy    = flag.String("eraser-pull-policy", string(corev1.PullIfNotPresent), "pull policy for eraser image")
```
can we handle this in helm chart instead of being a flag? perhaps similar to `AdditionalArgs` or a configmap we can read?

pmengelbert replied:
It's complicated because the configuration for the eraser/scanner/collector containers are set in code, and not in the helm chart. For example to get args to the scanner from the helm chart we have to set --scanner-arg on the manager.

pmengelbert replied:
The tests use `--set controllerManager.image.additionalArgs={...}` to propagate these values to the right place. Maybe we should consider a system-wide configmap that the manager looks at when it boots up, so that we don't clutter up the arguments like this? @sozercan if you agree I'll open an issue for that.

sozercan replied:
I am okay with configmap compared to cluttering arguments (which will get worse over time). Can we fix the e2e issue first then come up with a design for configmap solution?

pmengelbert replied:
The configmap issue is being tracked by #446. I'm addressing the e2e issues

pmengelbert committed:
Allow specifying namespace when getting logs in E2E test

The tests deployed by manifest use the `eraser-system` namespecace,
whereas the ones deployed by helm use a randomized test namespace.
Because of the way helm charts are generated, it's best to keep the
manifest test using `eraser-system`. This change allows that to happen
without breaking the rest of the tests.

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix github CI warning

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix other github CI warning

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

sozercan commented:
@pmengelbert do we want to rename the pr?

pmengelbert commented on main.go:
```diff
@@ -80,6 +85,21 @@ func main() {
+		*eraserPullPolicy:    &util.EraserPullPolicy,
+		*collectorPullPolicy: &util.CollectorPullPolicy,
+		*scannerPullPolicy:   &util.ScannerPullPolicy,
+	}
```
- [x] Flip these, the string keys could all have the same value.

pmengelbert committed:
Improve validation logic for pull policies

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove hardcoded namespaces in function with namespace parameter

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert commented on test/e2e/util/utils.go:
```diff
@@ -737,7 +767,7 @@ func CreateExclusionList(namespace string, list pkgUtil.ExclusionList) env.Func
 			ObjectMeta: metav1.ObjectMeta{
 				GenerateName: "excluded",
-				Namespace:    EraserNamespace,
+				Namespace:    TestNamespace,
```
This function accepts a `namespace` parameter, so this should be replaced with `namespace`

pmengelbert commented on test/e2e/util/utils.go:
```diff
@@ -747,7 +777,7 @@ func CreateExclusionList(namespace string, list pkgUtil.ExclusionList) env.Func
 		cMap := corev1.ConfigMap{}
 		err = wait.For(func() (bool, error) {
-			err := c.Resources().Get(ctx, excluded.Name, EraserNamespace, &cMap)
+			err := c.Resources().Get(ctx, excluded.Name, TestNamespace, &cMap)
```
here too

sozercan commented on controllers/util/util.go:
```diff
@@ -24,6 +25,10 @@ var (
 	EraserImage = flag.String("eraser-image", "ghcr.io/azure/eraser:latest", "eraser image")
 	EraserArgs  = utils.MultiFlag([]string{})
+
+	eraserPullPolicy    = flag.String("eraser-pull-policy", string(corev1.PullIfNotPresent), "pull policy for eraser image")
```
i thought we wanted to move these parts out of this PR?

pmengelbert replied:
@sozercan I misunderstood. Prior to this PR, the Pull Policy was hard coded so there is no option for the user (or tests) to change it. I thought you meant "leave the flags for now, then implement the configmap later and remove the flags". We need either the flags or the configmap for this PR. Should I put this one on hold until the configmap is implemented?

pmengelbert commented:
Closing this. See #458
